src/axiosCromwell.ts

```diff
@@ -1,10 +1,51 @@
 import axios from 'axios';
+import { useDispatch } from "react-redux";
+import { sessionExpired } from "./store/auth/actions";
 
 const axiosCromwell = axios.create({
-  baseURL: 'http://localhost:3001/',
+  baseURL: 'http://localhost:8080/',
   headers: {
     'Content-Type': 'application/json'
   }
 });
 
+axiosCromwell.interceptors.request.use(
+  config => {
+    const accessToken = localStorage.getItem('accessToken');
+    if (accessToken) {
+      config.headers["Authorization"] = accessToken;
+    }
+    return config;
+  },
+  error => {
+    Promise.reject(error);
+  }
+);
+
+axiosCromwell.interceptors.response.use(
+  response => {
+    return response;
+  },
+  error => {
+    const originalRequest = error.config;
+    let refreshToken = localStorage.getItem('refreshToken');
+    if (refreshToken && error.response.status === 401 && !originalRequest._retry) {
+      originalRequest._retry = true;
+      return axiosCromwell
+        .get('/auth/refresh', {params: {refreshToken: refreshToken}})
+        .then(response => {
```

Comment:
If you declare error handler as `async error =>  ...`, then you can `await` this axios call, and process in in a more linear fashion.

Reply:
And if you do this, you can rewrite `return Promise.reject(error)` as just `throw error` instead.

Reply:
Brilliant :gem: